vkcs/dataplatform/resource_cluster.go

```diff
@@ -247,6 +247,48 @@ func (r *clusterResource) Update(ctx context.Context, req resource.UpdateRequest
 		return
 	}
 
+	updateSettingsOpts := make([]clusters.ClusterUpdateSetting, 0)
+	planSettings, diags := resource_cluster.ExpandClusterConfigsSettings(ctx, plan.Configs.Settings)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	dataSettings, diags := resource_cluster.ExpandClusterConfigsSettings(ctx, data.Configs.Settings)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	for _, planSetting := range planSettings {
```

Comment:
this misses new and deleted settings

Reply:
Rewrote logic: apply changes according to plan. If planned settings are empty then do nothing and update state, if planned settings are non-empty then update their values and merge these settings with what came from api after update